crates/holochain/src/core/ribosome/guest_callback/validate.rs

```diff
@@ -276,6 +278,44 @@ mod slow_tests {
         assert_eq!(result, ValidateResult::Valid,);
     }
 
+    #[tokio::test(flavor = "multi_thread")]
+    async fn test_validate_implemented_invalid_return() {
+        let mut u = Unstructured::new(&NOISE);
+        let mut validate_invocation = ValidateInvocation::arbitrary(&mut u).unwrap();
+        let ribosome = RealRibosomeFixturator::new(Zomes(vec![TestWasm::ValidateInvalidReturn]))
+            .next()
+            .unwrap();
+        validate_invocation.zomes_to_invoke =
+            ZomesToInvoke::One(IntegrityZome::from(TestWasm::ValidateInvalidReturn).erase_type());
+
+        let err = ribosome
+            .run_validate(fixt!(ValidateHostAccess), validate_invocation)
+            .await
+            .unwrap_err();
+
+        let_assert!(RibosomeError::CallbackInvalidReturnType(err_msg) = err);
+        assert!(err_msg == "invalid value: integer `42`, expected variant index 0 <= i < 3");
```

Comment:
I just want to assert that the actual and expected types are given in the error. For general with tests I prefer to start with strict tests and then if someone does a small change in the future that breaks a bunch of tests then they can make the decision whether the tests were helpful during the change or too limiting and then change them if needed. Rather than assuming that we are going to change it in the future and we never do :-)